analyzers/src/SonarAnalyzer.Common/Rules/Utilities/AnalysisWarningAnalyzerBase.cs

```diff
@@ -42,18 +40,15 @@ public abstract class AnalysisWarningAnalyzerBase : UtilityAnalyzerBase
                 {
                     // This can be removed after we bump Microsoft.CodeAnalysis references to 3.0 or higher.
                     var path = Path.GetFullPath(Path.Combine(OutPath, "../../AnalysisWarnings.MsBuild.json"));
-                    lock (FileWriteLock)
```

Comment:
This one is questionable, because the path is not per-project, but shared per-solution.
If the analyzer is used to analyze two projects in parallel from the same application domain (not sure if possible from Roslyn side), it could end up in a race condition.

So the reason to remove this is that it doesn't make sense anyway. Because it will most likely be another project writhing this for 2nd time, not the same application domain. The `try/catch` should take care of conflicts.